src/Core/Field.h

```diff
@@ -681,6 +672,25 @@ class Field
 
     Types::Which which;
 
+    /// This function is prone to type punning and should never be used outside of Field class,
+    /// whenever it is used within this class the stored type should be checked in advance.
+    template <typename T>
+    NearestFieldType<std::decay_t<T>> & get()
```

Comment:
What about naming
- `get` as `get_unsafe` (or `unsafeGet`),
- and `safeGet` as `get`
?

Reply:
I thought about this (and it would be much easier refactoring), but I decided that it's better to left it more obvious so whoever uses it would be aware that get is not that simple...